File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int first_occ(vector<int> arr, int N, int ele)
    {
        int result,start,end,mid;
        start = 0;
        end = N-1;
        result = -1;
        while(start <= end)
        {
            mid = start + (end - start)/2;

            if(ele == arr[mid]) ///if found
            {
                result = mid;
                end = mid - 1;  ///search if any present in left portion
            }
            else if(ele < arr[mid])
            {
                end = mid - 1;
            }
            else
            {
                start = mid + 1;
            }
        }
        return result;
    }

    int last_occ(vector<int> arr, int N, int ele)
    {
        int result,start,end,mid;
        start = 0;
        end = N-1;
        result = -1;
        while(start <= end)
        {
            mid = start + (end - start)/2;

            if(ele == arr[mid]) ///if found
            {
                result = mid;
                start = mid + 1;    ///search if any present in right portion
            }
            else if(ele < arr[mid])
            {
                end = mid - 1;
            }
            else
            {
                start = mid + 1;
            }
        }
        return result;

    }
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> result;
        result.push_back(first_occ(nums, nums.size(), target));
        result.push_back(last_occ(nums, nums.size(), target));
        
        return result;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def firstOcc(self, nums, target):
        start = 0
        end = len(nums) - 1
        result = -1
        while start <= end:
            mid = start + (end - start) // 2
            if nums[mid] == target:
                result = mid
                end = mid - 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
        return result

    def lastOcc(self, nums, target):
        start = 0
        end = len(nums) - 1
        result = -1
        while start <= end:
            mid = start + (end - start) // 2
            if nums[mid] == target:
                result = mid
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
        return result

    def searchRange(self, nums: List[int], target: int) -> List[int]:
        fc = self.firstOcc(nums, target)
        lc = self.lastOcc(nums, target)
        result = [fc, lc]
        return result